ocaml/xapi-storage-script/main.ml

```diff
@@ -11,18 +11,293 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
  *)
-module U = Unix
 module R = Rpc
-module B = Backtrace
-open Core
-open Async
-open Xapi_storage_script_types
-module Plugin_client = Xapi_storage.Plugin.Plugin (Rpc_async.GenClient ())
-module Volume_client = Xapi_storage.Control.Volume (Rpc_async.GenClient ())
-module Sr_client = Xapi_storage.Control.Sr (Rpc_async.GenClient ())
-module Datapath_client = Xapi_storage.Data.Datapath (Rpc_async.GenClient ())
+module Types = Xapi_storage_script_types
+module Plugin_client = Xapi_storage.Plugin.Plugin (Rpc_lwt.GenClient ())
+module Volume_client = Xapi_storage.Control.Volume (Rpc_lwt.GenClient ())
+module Sr_client = Xapi_storage.Control.Sr (Rpc_lwt.GenClient ())
+module Datapath_client = Xapi_storage.Data.Datapath (Rpc_lwt.GenClient ())
 
-let ( >>>= ) = Deferred.Result.( >>= )
+let ( >>= ) = Lwt.bind
+
+let ( >>| ) = Fun.flip Lwt.map
+
+let ( >>>= ) = Lwt_result.bind
+
+let return = Lwt_result.return
+
+let fail = Lwt_result.fail
+
+let ( // ) = Filename.concat
+
+module Deferred = struct
+  let all_unit (lst : unit Lwt.t list) : unit Lwt.t =
+    let (_ : unit list Lwt.t) = Lwt.all lst in
+    Lwt.return ()
```

Comment:
this isn't exactly equivalent: it won't wait until all promises in 'lst' are resolved, it will instead return immediately.

Reply:
What you want here is Lwt.join